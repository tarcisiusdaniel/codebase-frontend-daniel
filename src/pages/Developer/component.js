import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { ROUTES } from '../../configs';

export default class Component extends React.Component {
  render() {
    const name = this.props.match.params.name;
    return (
      <section>
        <Typography>
          <Link to={ROUTES.HOME()}>Back to home</Link>
        </Typography>
        <Typography variant="display2">Developer name: {name}</Typography>
      </section>
    );
  }
}

Component.defaultProps = {
  match: {},
};

Component.propTypes = {
  match: PropTypes.object
};