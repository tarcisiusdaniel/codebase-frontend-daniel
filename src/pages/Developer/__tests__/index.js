import React from 'react';
import renderer from 'react-test-renderer';
import Developer from '../index';

jest.mock('react-router-dom');

describe('Developer', () => {
  it('renders correctly', () => {
    const match = {
      params: {
        name: 'Hello'
      }
    };
    const tree = renderer
      .create(<Developer match={match} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});