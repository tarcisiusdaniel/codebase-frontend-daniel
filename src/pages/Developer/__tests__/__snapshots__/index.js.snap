// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developer renders correctly 1`] = `
<section>
  <p
    className="MuiTypography-root-1 MuiTypography-body1-10"
  />
  <h1
    className="MuiTypography-root-1 MuiTypography-display2-4"
  >
    Developer name: 
    Hello
  </h1>
</section>
`;
